test/v1/controllers/requestParsers/ListBsasRequestDataParserSpec.scala

```diff
@@ -44,7 +44,7 @@ class ListBsasRequestDataParserSpec extends UnitSpec{
         MockValidator.validate(inputData).returns(Nil)
 
         parser.parseRequest(inputData) shouldBe
-          Right(ListBsasRequest(Nino(nino), DesTaxYear.fromMtd(taxYear), Some(typeOfBusiness), Some(selfEmploymentId)))
+          Right(ListBsasRequest(Nino(nino), DesTaxYear.fromMtd(taxYear), Some(incomeSourceIdentfier), Some(selfEmploymentId)))
       }
```

Comment:
There are lots of changes and quite specific logic that has gone into this Parser, I can't see the tests for this logic.